analyzer/src/traversal/expressions.rs

```diff
@@ -945,21 +950,27 @@ fn validate_are_kw_args(args: &[Node<fe::CallArg>]) -> Result<(), SemanticError>
     Ok(())
 }
 
-fn validate_is_numeric_literal(call_arg: &fe::CallArg) -> Result<String, SemanticError> {
+fn validate_is_numeric_literal(call_arg: &fe::CallArg) -> Result<BigInt, SemanticError> {
     let value = call_arg_value(call_arg);
 
     if let fe::Expr::UnaryOperation { operand, op: _ } = &value.kind {
         if let fe::Expr::Num(num) = &operand.kind {
-            return Ok(format!("-{}", num));
+            let num = numeric::Literal::new(num)
+                .parse::<BigInt>()
+                .map_err(|_| SemanticError::numeric_literal_expected())?;
+            return Ok(-num);
         }
     } else if let fe::Expr::Num(num) = &value.kind {
-        return Ok(num.to_string());
+        let num = numeric::Literal::new(num)
+            .parse::<BigInt>()
+            .map_err(|_| SemanticError::numeric_literal_expected())?;
```

Comment:
Should a parse failure here just panic? It'd be indicative of a bug in the lexer/parser, right?

Reply:
Exactly! I fixed it.